include/seqan/parallel/parallel_atomic_primitives.h

```diff
@@ -231,9 +231,13 @@ struct Atomic
 // Implementation in MSVC
 // ----------------------------------------------------------------------------
 
+// NOTE(marehr): clang/c2 v3.7 doesn't know #pragma intrinsic.
+#if !defined(COMPILER_CLANG)
```

Comment:
if this only affects c2 and not the unix clang, than please use `!defined(__c2__)`